src/Sylius/Bundle/AdminBundle/Resources/config/grids/product.yml

```diff
@@ -44,6 +44,14 @@ sylius_grid:
                 enabled:
                     type: boolean
                     label: sylius.ui.enabled
+                channel:
+                    type: entities
+                    label: sylius.ui.channel
+                    form_options:
+                        class: "%sylius.model.channel.class%"
+                    options:
+                        class: "%sylius.model.channel.class%"
+                        field: "channels.id"
```

Comment:
Does it work without changes to ProductRepository?

Reply:
@pamil what do you think about this solution?

Reply:
Yes, it does! If it works, it's much better to edit only one place instead of two places.

Reply:
yes, product repository is fine, we dont need to add anything there 💃